### DISCUSSION TITLE
Force component to re-render (like Livewire $refresh)

### DISCUSSION THREAD

**Discussion by jacksleight** [alpinejs/alpine#4319] [1. Help] [answered]
### Problem

Livewire has a `$refresh` method that forces the component to re-render, is there any equivalent in Alpine?

My use case is I have an element that is toggled with `x-show`, but part of the condition is based on the viewport size. By default nothing changes here when the window is resized:

```html
<div
	x-data
	x-show="someConditionInvolvingViewportSize">
</div>
```

I thought it might be possible to use the new Resize plugin to trigger a re-render, but it don't think anything like this exists:

```html
<div
	x-data
	x-resize.document="$refresh()"
	x-show="someConditionInvolvingViewportSize">
</div>
<!-- My condition uses `window.matchMedia`, which is why I don't need the dimensions provided by Resize -->
```

Is there another way?

### Workaround

I found this works, but it would be great if there was a more official way to do it:

```html
<div
	x-data="{ tick: 1 }"
	x-resize.document="tick++"
	x-show="tick && someConditionInvolvingViewportSize">
</div>
```

**Comment by SimoTod** [comment, marked as answer]
You need observable variables (something from x-data or something using [Alpine.reactive](https://alpinejs.dev/advanced/reactivity#alpine-reactive)) to have reactive directives in Alpine so the correct way to do it is

```html
<div
	x-data="{ show: false }"
	x-resize.document="show = someConditionInvolvingViewportSize"
	x-show="show">
  ok
</div>
```

**Reply by SimoTod** [reply]
About the original question: no, there's nothing in Alpine for refreshing a component on demand.

**Reply by jacksleight** [reply]
Ah gotcha, that makes much more sense. Thanks!

**Reply by SimoTod** [reply]
No worries, just for completeness (and for those who are going to read the discussion in the future): Unless your condition is really complex and involves other stuff in addition to the viewport size, you can use a CSS rule to do it and it's usually more performant. 👍

**Reply by ekwoka** [reply]
If 100% unavoidable, you can `Alpine.destroyTree` then `Alpine.initTree` but I can be pretty confident in saying that this should never be the case.

(I mean, I've done it once, but it was for an Alpine Sandbox when you change the code...)
